### ISSUE TITLE
Block switcher disabled state should be consistent with other buttons disabled state

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51445]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/49865 / https://github.com/WordPress/gutenberg/pull/50944

In many places in the UI we do use a pattern for disabled buttons to keep them focusable and discoverable. Most of these buttons aren't 'truly' disabled. Instead, they get an `aria-disabled="true"` attribute, they're noop'ed, and they visually look disabled. For more details, see [Focusability of disabled controls in the ARIA practices](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols).

The block switcher button doesn't follow this pattern. It should. This button can be in a 'disabled' state based on a few conditions, for example when a block is locked. In all cases it should still be focusable.

Actually, the block switcher is truly disabled by the means of a `disabled` attribute. Also, it doesn't visually look disabled. as such:
- The button is not focusable: screen reader users can't tab to this button and they will miss important information about the block type.
- Visually, this button doesn't look disabled. While I do understand it is used as a visual clue for the block type, the disabled state should be communicated also visually.

Note: the visual disabled state should work also when the preference 'Show button text labels' is enabled..

Screenshot of a locked paragraph block:
- The block switcher button looks enabled, while it is actually disabled.
- It isn't focusable so that screen reader users will miss the block type information provided by this button.
- Notice that, for example, the Undo and Redo buttons in the top bar do have a disabled style.

<img width="1118" alt="Screenshot 2023-06-13 at 10 08 53" src="https://github.com/WordPress/gutenberg/assets/1682452/7b28cf08-81b5-4793-bec7-28c6354da0fe">

Screenshot: with the 'Show button text labels' preference enabled:
- Even more confusing maybe, as the style of the disabled block switcher is identical to the style of all the other buttons in the block toolbar.
- Notice that, for example, the Undo and Redo buttons in the top bar do have a disabled style also when they show a textual label.

<img width="1159" alt="Screenshot 2023-06-13 at 10 09 11" src="https://github.com/WordPress/gutenberg/assets/1682452/297c21c5-7971-4f71-a15a-7a6286a240d7">

### Step-by-step reproduction instructions

- Go to the post editor.
- Add a Paragraph block, add some text, and lock the block.
- The block switcher button is now 'truly' disabled.
- Observe the button doesn't have a disabled style.
- Use the Tab key and move to the block toolbar.
- Use the Right and Left arrow keys to navigate through the block toolbar buttons.
- Observe that is not possible to move focus to the block switcher button.
- Screen reader users will miss the block type information provided by the block switcher button.
- Go to the editor Options > Preferences, and enable 'Show button text labels'.
- Repeat the steps above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
We can probably resolve this via #57892.
